C-Plus-Plus/ds/avl_tree.cpp

```diff
@@ -0,0 +1,481 @@
+    preorder_traversal(root);
+    cout<<"\n";
+
+    int m;//number of elements to be deleted
+    cout<<"\nEnter the number of nodes to be deleted: ";
+    cin>>m;
+    cout<<"Enter the node values to be deleted: ";
+    vector<int>deletion_order(m);
+    for(int i=0;i<deletion_order.size();i++){
+        cin>>deletion_order[i];
+        root=delete_node(root,deletion_order[i],root->data);
+        cout<<"\nafter deleting node with key value = "<<deletion_order[i]<<"\n";
+        preorder_traversal(root);
+        cout<<"\n";
+    }
+    return 0;
+}
+/*
+Test Case 1:
+  (insertion)
+
+  Input:
+  13
+  14 17 11 7 53 4 13 12 8 60 19 16 20
+  Output(preorder traversal):
+  14(0) 11(0) 7(0) 4(0) 8(0) 12(-1) 13(0) 19(0) 17(1) 16(0) 53(0) 20(0) 60(0)
+
+  (deletion)
+  Input:
+  5
+  14 13 19 17 12
+  Output:
+  after deleting node with key value = 14
+    13(0) 11(1) 7(0) 4(0) 8(0) 12(0) 19(0) 17(1) 16(0) 53(0) 20(0) 60(0)
+
+  after deleting node with key value = 13
+    12(0) 7(-1) 4(0) 11(1) 8(0) 19(0) 17(1) 16(0) 53(0) 20(0) 60(0)
+
+  after deleting node with key value = 19
+    12(0) 7(-1) 4(0) 11(1) 8(0) 17(-1) 16(0) 53(0) 20(0) 60(0)
+
+  after deleting node with key value = 17
+    12(0) 7(-1) 4(0) 11(1) 8(0) 53(1) 16(-1) 20(0) 60(0)
+
+  after deleting node with key value = 12
+    11(-1) 7(1) 4(0) 53(1) 16(-1) 20(0) 60(0)
+*/
+/*
+The AVL Tree example used in this code:
+                       14
+                /            \
+               11             19
+              /   \         /    \
+             7     12       17    53
+            / \      \     /      / \
+           4   8     13   16    20   60
+
+
+after deleting 14 it should look like this:
+                     13
+                /            \
+               11             19
+              /   \         /    \
+             7     12      17    53
+            / \            /     / \
+           4   8          16   20   60
+
+after deleting 13 it should look like this:
+                       12
+                   /        \
+                7             19
+              /   \         /    \
+            4      11      17    53
+                   /       /     / \
+                  8       16   20   60
+
+after deleting 19 it should look like this:
+                        12
+                    /        \
+                  7           17
+                /   \        /  \
+               4      11    16   53
+                     /          /  \
+                    8          20   60
+
+after deleting 17 it should look like this:
+                        12
+                      /    \
+                     7      53
+                   /  \     / \
+                  4   11   16  60
+                     /       \
+                    8         20
+
+ */
+//Other cases that you could try:
+//vector<int>data={10,20,30,40,50,25};
+//vector<int>data={33,13,53,9,21,61,8,11};
+//vector<int>data={14,13};
```

Comment:
Add a new line here.

Reply:
sir, I am not sure I understand. Are you saying to add a blank line after "Other cases that you could try:" ?

Reply:
Add a blank at the end of the file.